### DISCUSSION TITLE
Right way to publish authors on PyPi from setuptools

### DISCUSSION THREAD

**Discussion by luclaurent** [pypa/setuptools#3800] [Q&A] [answered]
I currently use `setuptools` to build my Python's package and I have declared the two authors that way on my `pyproject.toml` file:

    authors = [
         {name = "X Y", email = "x.y@tt.net"},
         {name = "Z H", email = "z.h@tt.net"},
    ]

Everything works and I can publish it on PyPi but only the first author is published. How can I display both authors? 

I have tried to use the following syntax

    authors = ["X Y <x.y@tt.net>, Z H <z.h@tt.net>"]

But I have the following error

    ValueError: invalid pyproject.toml config: `project.authors[{data__authors_x}]`.
    configuration error: `project.authors[{data__authors_x}]` must be object

Notice that I specify:

    [build-system]
    requires = ["setuptools","numpy","scipy","wheel"]
    build-backend = "setuptools.build_meta"

**Comment by abravalheri** [comment, marked as answer]
Hi @luclaurent, if you unzip the generated wheel, and have a look on the `*.dist-info/METADATA` file, how does the `Author-email` field show up?

If all the authors are listed there correctly[^1], it might be worth contacting PyPI teams to understand which is the logic they use to display authors...

---

I am not sure if this will work, but I think you could try something like:

```toml
authors = [{name = "X Y <x.y@tt.net>, Z H <z.h@tt.net>"}]
```

[^1]: Please note that `setuptools` follows the [PEP 621 specification](https://peps.python.org/pep-0621/#authors-maintainers), which requires `{name = ..., email = ...}` to be transformed in [core metadata's author-email field](https://packaging.python.org/en/latest/specifications/core-metadata/#author). PyPI uses this Core Metadata to populate its UI according to their own custom logic. You can change the way you write the `pyproject.toml` file to manipulate how the `Author` and `Author-email` fields are written in the `METADATA` file, but in the end of the day how PyPI decides to show them is something we don't have much control over.

**Reply by webknjaz** [reply]
My best guess is that the display bit in warehouse is missing...

**Reply by luclaurent** [reply]
I confirm that the file `*.dist-info/METADATA` contains the right `Author-email` information. The issue is from PyPI's side...

**Reply by webknjaz** [reply]
@luclaurent FYI the tracker is here: https://github.com/pypi/warehouse/issues.
